pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -549,22 +584,24 @@ func (*SQLPlugin) GetPluginInfo(context.Context, *spi.GetPluginInfoRequest) (*sp
 }
 
 func (ds *SQLPlugin) withWriteRepeatableReadTx(ctx context.Context, op func(tx *gorm.DB) error) error {
-	return ds.withTx(ctx, op, false, &sql.TxOptions{Isolation: sql.LevelRepeatableRead})
+	return ds.withTx(ctx, ds.db, op, false, &sql.TxOptions{Isolation: sql.LevelRepeatableRead})
 }
 
 func (ds *SQLPlugin) withWriteTx(ctx context.Context, op func(tx *gorm.DB) error) error {
-	return ds.withTx(ctx, op, false, nil)
+	return ds.withTx(ctx, ds.db, op, false, nil)
 }
 
 func (ds *SQLPlugin) withReadTx(ctx context.Context, op func(tx *gorm.DB) error) error {
-	return ds.withTx(ctx, op, true, nil)
-}
-
-func (ds *SQLPlugin) withTx(ctx context.Context, op func(tx *gorm.DB) error, readOnly bool, opts *sql.TxOptions) error {
+	if ds.roDb != nil {
+		return ds.withTx(ctx, ds.roDb, op, true, nil)
+	}
```

Comment:
Obviously, we need to enable use of the read-only connection to test any performance gains. However, pending resolution of open questions about consistency-sensitive operations in #1239, we should probably leave this using the read/write connection in this commit. It might not wind up being this simple if we need special logic for some reads not to use r/o.